### DISCUSSION TITLE
How to change the locale startOfweek ?

### DISCUSSION THREAD

**Discussion by abdullah-cse** [jama5262/jiffy#272] [Q&A] [answered]
Thanks for awesome Package!😍 It's made my life easier. I'm only struggling about how to change **localeStartOfWeek** manually.

```dart
  //Getting default Start Date of Week
  var defaultStartOfWeek = Jiffy.now().localeStartOfWeek;
  print(defaultStartOfWeek); //StartOfWeek.sunday
  
  //Trying to change Start Date of Week manually
  defaultStartOfWeek = StartOfWeek.monday;

  var xyz = Jiffy.now();
  //Expected Output= StartOfWeek.monday
  print(xyz.localeStartOfWeek); //StartOfWeek.sunday

```
Here in this snippet, expecting `StartOfWeek.monday` but getting `StartOfWeek.sunday`

### So, why i need this as user can get startOfWeek by their locale?
If user want to change First date of week manually.

So how to change `StartOfWeek` manually?

**Comment by jama5262** [comment]
Hey @abdullah-cse 

Glad you like `Jiffy`

And yes, we should have a way to manually set the start day of week, since its really in demand

Will work on it, and see how first I can ship this feature out

**Comment by jama5262** [comment, marked as answer]
As of version 6.4.0, you can now [override locale settings](https://github.com/jama5262/jiffy/tree/v6.4.0/doc#advanced-locale-setting)

So to [override the start of week](https://github.com/jama5262/jiffy/tree/v6.4.0/doc#start-of-week), simply do the following

```dart
await Jiffy.setLocale(locale: 'en_US');

final jiffy = Jiffy.now().locale.startOfWeek; // Sunday

// Overriding the start of week
await Jiffy.setLocale(
    locale: 'en_US',
    startOfWeek: StartOFWeek.saturday,
);

final jiffy = Jiffy.now().locale.startOfWeek; // Saturday
```
